gapic/src/main/com/google/gapid/views/TextureView.java

```diff
@@ -119,99 +80,42 @@ public TextureView(Composite parent, Models models, Widgets widgets) {
     imagePanel.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
 
     imagePanel.createToolbar(toolBar, widgets.theme);
     gotoAction.createToolItem(toolBar);
 
     models.capture.addListener(this);
-    models.commands.addListener(this);
-    models.resources.addListener(this);
     models.follower.addListener(this);
     addListener(SWT.Dispose, e -> {
       models.capture.removeListener(this);
-      models.commands.removeListener(this);
-      models.resources.removeListener(this);
       models.follower.removeListener(this);
       gotoAction.dispose();
-      imageProvider.reset();
-    });
-
-    ViewerFilter filterDeleted = new ViewerFilter() {
-      @Override
-      public boolean select(Viewer viewer, Object parentElement, Object element) {
-        return !((Data)element).deleted;
-      }
-    };
-
-    textureTable.getTable().addListener(SWT.Selection, e -> updateSelection());
-    showDeleted.addListener(SWT.Selection, e -> {
-      if (showDeleted.getSelection()) {
-        textureTable.removeFilter(filterDeleted);
-      } else {
-        textureTable.addFilter(filterDeleted);
-      }
     });
+    reinitialize();
   }
 
   protected void setImage(FetchedImage result) {
     imagePanel.setImage(result);
   }
 
-  private static void initTextureSelector(TableViewer viewer, ImageProvider imageProvider) {
-    viewer.setContentProvider(ArrayContentProvider.getInstance());
-    viewer.setLabelProvider(new ViewLabelProvider(imageProvider));
-
-    sorting(viewer,
-        createTableColumn(viewer, "Type", Data::getType,
-            Comparator.comparing(Data::getType)),
-        createTableColumn(viewer, "ID", Data::getId, d -> imageProvider.getImage(d),
-            (d1, d2) -> UnsignedLongs.compare(d1.getSortId(), d2.getSortId())),
-        createTableColumn(viewer, "Name", Data::getLabel,
-            Comparator.comparing(Data::getLabel)),
-        createTableColumn(viewer, "Width", Data::getWidth,
-            Comparator.comparingInt(Data::getSortWidth)),
-        createTableColumn(viewer, "Height", Data::getHeight,
-            Comparator.comparingInt(Data::getSortHeight)),
-        createTableColumn(viewer, "Depth", Data::getDepth,
-            Comparator.comparingInt(Data::getSortDepth)),
-        createTableColumn(viewer, "Layers", Data::getLayers,
-            Comparator.comparingInt(Data::getSortLayers)),
-        createTableColumn(viewer, "Levels", Data::getLevels,
-            Comparator.comparingInt(Data::getSortLevels)),
-        createTableColumn(viewer, "Format", Data::getFormat,
-            Comparator.comparing(Data::getFormat)));
-  }
-
   @Override
   public Control getControl() {
     return this;
   }
 
   @Override
   public void reinitialize() {
-    onCaptureLoadingStart(false);
-    updateTextures(true);
+    if (!models.capture.isLoaded()) {
+      onCaptureLoadingStart(false);
+    } else {
+      onCaptureLoaded(null);
+    }
```

Comment:
FYI: This doesn't restore the currently selected texture, which is why I think that should be tracked in the model, rather than via the follow.

Reply:
Tracked in the model as discussed.